### ISSUE TITLE
Ingressgateway readiness probes failing with 503. Envoy proxy is NOT ready: config not received from Pilot (is Pilot running?) 

### ISSUE THREAD

**Issue by krackjack29** [istio/istio#27163]
**Bug description**
We have multiple ingress gateways deployed using the istio operator on a GKE cluster with multiple node pools and workload identity enabled (after google's suggestion that it would work on open source version of istio)

Istio Proxies (ingress gateways and egress gateways) unable to connect to istiod, readiness probes keep failing 
Envoy proxy is NOT ready: config not received from Pilot (is Pilot running?) 

The Istio control plane is deployed using the following IstioOperator 
```yaml
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
metadata:
  namespace: istio-system
  name: profile-demo-customized  
spec:
  hub: <redacted>/istio
  tag: 1.5.1
  profile: demo
  # Traffic management feature
  components:
    base:
      enabled: true
    pilot:
      enabled: true
      k8s:
        replicaCount: 1 # Due to the defect https://github.com/istio/istio/issues/22463, which will be resolved in 1.5.6
        env:
          - name: POD_NAME
            valueFrom:
              fieldRef:
                apiVersion: v1
                fieldPath: metadata.name
          - name: POD_NAMESPACE
            valueFrom:
              fieldRef:
                apiVersion: v1
                fieldPath: metadata.namespace
          - name: GODEBUG
            value: gctrace=1
          - name: PILOT_TRACE_SAMPLING
            value: "100"
          - name: CONFIG_NAMESPACE
            value: istio-config
        resources:
          requests:
            cpu: 10m
            memory: 100Mi
        overlays:
        - apiVersion: policy/v1beta1
          kind: PodDisruptionBudget
          name: istiod
          patches:
          - path: spec.selector.matchLabels
            value: 
              app: istiod
              istio: pilot
    # Policy feature
    policy:
      enabled: true
      k8s:
        replicaCount: 2
   # Telemetry feature
    telemetry:
      enabled: true
      k8s:
        env:
          - name: POD_NAMESPACE
            valueFrom:
              fieldRef:
                apiVersion: v1
                fieldPath: metadata.namespace
          - name: GOMAXPROCS
            value: "6"
        hpaSpec:
          maxReplicas: 5
          minReplicas: 1
          scaleTargetRef:
            apiVersion: apps/v1
            kind: Deployment
            name: istio-telemetry
          metrics:
            - type: Resource
              resource:
                name: cpu
                targetAverageUtilization: 80
        replicaCount: 1
        resources:
          requests:
            cpu: 100m
            memory: 100Mi
          limits:
            cpu: 4800m
            memory: 4G
        strategy:
          rollingUpdate:
            maxSurge: "100%"
            maxUnavailable: "25%"

  # Config management feature
    galley:
      enabled: false

  # Auto injection feature
    sidecarInjector:
      enabled: false

  # Istio Gateway feature
    ingressGateways:
    - name: istio-ingressgateway
      enabled: false
     
    egressGateways:
    - name: istio-egressgateway
      enabled: true
      k8s:
        tolerations:
        - effect: NoSchedule
          key: workloadType
          operator: Equal
          value: gateway
        env:
          - name: ISTIO_META_ROUTER_MODE
            value: "sni-dnat"
        service:
          ports:
            - port: 80
              name: http2
            - port: 443
              name: https
            - port: 15443
              targetPort: 15443
              name: tls
        hpaSpec:
          maxReplicas: 5
          minReplicas: 1
          scaleTargetRef:
            apiVersion: apps/v1
            kind: Deployment
            name: istio-egressgateway
          metrics:
            - type: Resource
              resource:
                name: cpu
                targetAverageUtilization: 80
        resources:
          requests:
            cpu: 100m
            memory: 128Mi
          limits:
            cpu: 2000m
            memory: 1024Mi
        strategy:
          rollingUpdate:
            maxSurge: "100%"
            maxUnavailable: "25%"
    # Istio CNI feature
    cni:
      enabled: false

  addonComponents:
    prometheus:
      enabled: false
    kiali:
      enabled: true
    grafana:
      enabled: false
    tracing:
      enabled: true
    istiocoredns:
      enabled: false
  
  values:
    global:
      defaultNodeSelector:
        kubernetes.io/os: linux
      outboundTrafficPolicy:
        mode: REGISTRY_ONLY
      disablePolicyChecks: false
      mtls:
        enabled: true
      proxy:
        accessLogFile: /dev/stdout
        resources:
          limits: 
            cpu: 750m #Istio benchmark mentions the CPU consumption for 1000Requests per second
            memory: 512Mi 
      jwtPolicy: third-party-jwt
      pilotCertProvider: istiod
      controlPlaneSecurityEnabled: false 
    kiali:
      createDemoSecret: true
      dashboard:
        grafanaURL: http://grafana:3000
        secretName: kiali-admin
      prometheusAddr: http://prometheus-server.monitoring-system:9090
      hub: <redacted>/kiali
    tracing:
      jaeger:
        hub: <redacted>/jaegertracing
      zipkin:
        hub: <redacted>/openzipkin
```
while the multiple ingress gateways (1 nodeport exposed via GKE ingress and the other an internal load balancer) is deploying using the following IstioOperator

```yaml
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
metadata:
  name: istio-ingress-cluster-admin
  namespace: istio-system
spec:
  components:
    ingressGateways:
    - enabled: true
      k8s:
        env:
        - name: ISTIO_META_ROUTER_MODE
          value: sni-dnat
        hpaSpec:
          maxReplicas: 100
          metrics:
          - resource:
              name: cpu
              targetAverageUtilization: 80
            type: Resource
          minReplicas: 1
          scaleTargetRef:
            apiVersion: apps/v1
            kind: Deployment
            name: istio-ingressgateway
        nodeSelector:
          workloadType: gateway
        resources:
          limits:
            cpu: 2000m
            memory: 1024Mi
          requests:
            cpu: 100m
            memory: 128Mi
        service:
          ports:
          - name: http2
            port: 80
          - name: https
            port: 443
          - name: tls
            port: 15443
            targetPort: 15443
          type: NodePort
        serviceAnnotations:
          cloud.google.com/neg: '{"ingress": true}'
        strategy:
          rollingUpdate:
            maxSurge: 100%
            maxUnavailable: 25%
        tolerations:
        - effect: NoSchedule
          key: workloadType
          operator: Equal
          value: gateway
      name: istio-ingressgateway
    - enabled: true
      k8s:
        env:
        - name: ISTIO_META_ROUTER_MODE
          value: sni-dnat
        hpaSpec:
          maxReplicas: 50
          metrics:
          - resource:
              name: cpu
              targetAverageUtilization: 80
            type: Resource
          minReplicas: 1
          scaleTargetRef:
            apiVersion: apps/v1
            kind: Deployment
            name: istio-proxy-ingressgateway
        nodeSelector:
          workloadType: gateway
        overlays:
        - kind: HorizontalPodAutoscaler
          name: istio-proxy-ingressgateway
          patches:
          - path: metadata.labels.app
            value: istio-proxy-ingressgateway
          - path: metadata.labels.istio
            value: proxy-ingressgateway
          - path: spec.scaleTargetRef.name
            value: istio-proxy-ingressgateway
        - kind: Deployment
          name: istio-proxy-ingressgateway
          patches:
          - path: metadata.labels.app
            value: istio-proxy-ingressgateway
          - path: metadata.labels.istio
            value: proxy-ingressgateway
          - path: spec.selector.matchLabels.app
            value: istio-proxy-ingressgateway
          - path: spec.selector.matchLabels.istio
            value: proxy-ingressgateway
          - path: spec.template.metadata.labels.app
            value: istio-proxy-ingressgateway
          - path: spec.template.metadata.labels.istio
            value: proxy-ingressgateway
        - kind: Service
          name: istio-proxy-ingressgateway
          patches:
          - path: metadata.labels.app
            value: istio-proxy-ingressgateway
          - path: metadata.labels.istio
            value: proxy-ingressgateway
          - path: spec.selector.app
            value: istio-proxy-ingressgateway
          - path: spec.selector.istio
            value: proxy-ingressgateway
        - kind: Gateway
          name: ingressgateway
          patches:
          - path: metadata.name
            value: proxy-ingressgateway
          - path: spec.selector.istio
            value: proxy-ingressgateway
        - kind: PodDisruptionBudget
          name: ingressgateway
          patches:
          - path: metadata.name
            value: proxy-ingressgateway
          - path: metadata.labels.app
            value: istio-proxy-ingressgateway
          - path: metadata.labels.istio
            value: proxy-ingressgateway
          - path: spec.selector.matchLabels.app
            value: istio-proxy-ingressgateway
          - path: spec.selector.matchLabels.istio
            value: proxy-ingressgateway
        resources:
          limits:
            cpu: 2000m
            memory: 1024Mi
          requests:
            cpu: 100m
            memory: 128Mi
        service:
          ports:
          - name: http2
            port: 80
          - name: https
            port: 443
          - name: tls
            port: 15443
            targetPort: 15443
        serviceAnnotations:
          cloud.google.com/load-balancer-type: internal
        strategy:
          rollingUpdate:
            maxSurge: 100%
            maxUnavailable: 25%
        tolerations:
        - effect: NoSchedule
          key: workloadType
          operator: Equal
          value: gateway
      name: istio-proxy-ingressgateway
  hub: <redacted>/istio
  profile: empty
  tag: 1.5.1
```
This setup was working fine until we upgraded our cluster to workload identity and GKE Metadata server based on Google's recommendation. Now the proxies (including ingress gateways and egress gateways) readiness probes are failing with 503 errors. 

**Istioctl proxy-status **
![image](https://user-images.githubusercontent.com/30447663/92565585-821b4f00-f298-11ea-97ac-496c558d0656.png)

**Note** if I delete the ingress gateway istio operator, egress gateway and other proxies will start connecting to istiod and will start to work well. Same configuration works well without workload identity.

**Affected product area (please put an X in all that apply)**

[ ] Docs
[ ] Installation
[x] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Expected behavior**
Proxies should be able to get the LDS,RDS from istiod and the readiness probe should return 200.

**Version (include the output of `istioctl version --remote` and `kubectl version --short` and `helm version` if you used Helm)**
### Istio Version
client version: 1.6.0
policy version: 1.5.1
policy version: 1.5.1
proxy-ingressgateway version:
telemetry version: 1.5.1
pilot version: 1.5.1
data plane version: 1.5.1 (5 proxies)

### Kubectl version
Client Version: v1.17.3
Server Version: v1.15.12-gke.2

**How was Istio installed?**
Using the IstioOperator, configuration provided above

**Environment where bug was observed (cloud vendor, OS, etc)**
GKE

**Comment by howardjohn** [comment]
Can you provide logs from the pilot and gateway pods? it should show why its not ready. The full logs, not just the 1 line snippet. Thanks!

**Comment by krackjack29** [comment]
> Can you provide logs from the pilot and gateway pods? it should show why its not ready. The full logs, not just the 1 line snippet. Thanks!

Attached are the logs. 
[ingress-gateway.log](https://github.com/istio/istio/files/5196234/ingress-gateway.log)
[istiod.log](https://github.com/istio/istio/files/5196237/istiod.log)

Whats interesting is one of the workload test pod that I have enabled is throwing up citadelclient error

error citadelclient Failed to create certificate: rpc error: code = Unavailable desc = connection error: desc = "transport: authentication handshake failed: x509: certificate signed by unknown authority (possibly because of \"crypto/rsa: verification error\" while trying to verify candidate authority certificate \"cluster.local\")"

**Comment by howardjohn** [comment]
Any chance you can update Istio? There was some related PR that may be the cause: https://github.com/istio/envoy/commit/fd7d4a72b736021a3866cf7065ca4dc7f897b3b7

Preferably to 1.7, but 1.5.10 could work. SEe https://istio.io/latest/news/support/announcing-1.5-eol-final/

**Comment by krackjack29** [comment]
@howardjohn We would look to upgrade and see if it resolves. Thanks.

**Comment by krackjack29** [comment]
Quick update we tried upgrading to istio 1.6.1 and found better logs, at least in the context. 

`gRPC config for type.googleapis.com/envoy.api.v2.Listener rejected: Error add
ing/updating listener(s) 0.0.0.0_443: error adding listener '0.0.0.0:443': multiple filter chains with the same matching rules are defined`

**Comment by howardjohn** [comment]
@krackjack29 can you give more info about the gateways/VS you have defined so we can reproduce and fix this? Also if you can get a config dump: https://github.com/istio/istio/wiki/Troubleshooting-Istio.

Thanks!

**Comment by howardjohn** [comment]
ping, any more info?

**Comment by krackjack29** [comment]
Hi @howardjohn apologies for not replying earlier. The issue was with duplicate service entries with different protocols for the same host.

We are in a multitenant environment where the cluster admins were adding service entries directly to the namespaces using GitOps and were not using exportTo functionality, by default the service entries were applicable to all namespaces and they had no idea that same host was added in another namespace using TLS, while they were adding HTTPS for same host.

We have streamlined the process to add service entries using exportTo: '.' to specific namespace. and control it using the OPA policies.  

I am closing this issue.
